detection_rules/rule_formatter.py

```diff
@@ -206,5 +206,5 @@ def _do_write(_data, _contents):
             _do_write(data, _contents)
 
     finally:
-        if needs_close:
+        if needs_close and isinstance(outfile, io.IOBase):
```

Comment:
we can loosen it up further
```suggestion
        if needs_close and hasattr(outfile, "close"):
```